src/neg_multiply.rs

```diff
@@ -0,0 +1,59 @@
+use rustc::hir::*;
+use rustc::lint::*;
+use syntax::codemap::{Span, Spanned};
+
+use consts::{self, Constant};
+use utils::span_lint;
+
+/// **What it does:** Checks for multiplication by -1 as a form of negation.
+///
+/// **Why is this bad?** It's more readable to just negate.
+///
+/// **Known problems:** This assumes that the type can be negated.
+///
+/// **Example:** `x * -1`
+declare_lint! {
+    pub NEG_MULTIPLY,
+    Warn,
+    "Warns on multiplying integers with -1"
+}
+
+#[derive(Copy, Clone)]
+pub struct NegMultiply;
+
+impl LintPass for NegMultiply {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(NEG_MULTIPLY)
+    }
+}
+
+#[allow(match_same_arms)]
+impl LateLintPass for NegMultiply {
+    fn check_expr(&mut self, cx: &LateContext, e: &Expr) {
+        if let ExprBinary(Spanned { node: BiMul, .. }, ref l, ref r) = e.node {
+            match (&l.node, &r.node) {
+                (&ExprUnary(..), &ExprUnary(..)) => (),
+                (&ExprUnary(ref neg, ref lit), _) =>
+                    if neg == &UnNeg { check_mul(cx, e.span, lit, r) },                
```

Comment:
Yes. I just wanted to show match guards. :smile: But I guess I won't need them here.